src/aof.c

```diff
@@ -1543,17 +1549,18 @@ int loadAppendOnlyFiles(aofManifest *am) {
         serverAssert(am->base_aof_info->file_type == AOF_FILE_TYPE_BASE);
         aof_name = (char*)am->base_aof_info->file_name;
         updateLoadingFileName(aof_name);
-
+        ++aof_num;
         start = ustime();
         ret = loadSingleAppendOnlyFile(aof_name);
-        if (ret == AOF_OK) {
+        if (ret == AOF_OK || (ret == AOF_TRUNCATED && aof_num == total_num)) {
             serverLog(LL_NOTICE, "DB loaded from base file %s: %.3f seconds",
                 aof_name, (float)(ustime()-start)/1000000);
+            ret = AOF_OK;
```

Comment:
i think it's odd to hide (override) the TRUNCATED status, let's propagate it to the caller when it's valid.
and override it with FAILED when it's not in the last file.
we'll need to update the two callers of that function to handle that value correctly.
debug.c isn't expecting it, so it can consider it as failure (maybe revert your change there?)
and server.c can consider it as success (which it already does AFAICT).

Reply:
OK